src/page/Product/components/productWishButton.tsx

```diff
@@ -0,0 +1,42 @@
+import { useSuspenseQuery } from '@tanstack/react-query';
+import { requests } from '@/api/requests';
+import { useWishToggleMutation } from '../hooks/useWishToggleMutation';
+
+interface ProductWishButtonProps {
+  index: number;
+}
+
+const ProductWishButton = ({ index }: ProductWishButtonProps) => {
+  const { data } = useSuspenseQuery({
+    queryKey: ['productWishData', index],
+    queryFn: () => requests.fetchProductWish(index),
+  });
+
+  const { mutate: toggleWish } = useWishToggleMutation(index);
+
+  const handleWishClick = () => {
+    toggleWish();
+  };
+
+  return (
+    <button onClick={handleWishClick}>
```

Comment:
```suggestion
  return (
    <button onClick={toggleWish}>
```
요로코롬 바로 toogleWish를 사용해도 무방해보여요!